File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-com-5-4/CommodityMetalBrand.cpp
// CommodityMetalBrand.cpp 
#include "CommodityMetalBrand.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

CommodityMetalBrand::CommodityMetalBrand(TiXmlNode* xmlNode)
: ISerialized(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //nameNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* nameNode = xmlNode->FirstChildElement("name");

   if(nameNode){nameIsNull_ = false;}
   else{nameIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- nameNode , address : " << nameNode << std::endl;
   #endif
   if(nameNode)
   {
      if(nameNode->Attribute("href") || nameNode->Attribute("id"))
      {
          if(nameNode->Attribute("id"))
          {
             nameIDRef_ = nameNode->Attribute("id");
             name_ = boost::shared_ptr<CommodityMetalBrandName>(new CommodityMetalBrandName(nameNode));
             name_->setID(nameIDRef_);
             IDManager::instance().setID(nameIDRef_,name_);
          }
          else if(nameNode->Attribute("href")) { nameIDRef_ = nameNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { name_ = boost::shared_ptr<CommodityMetalBrandName>(new CommodityMetalBrandName(nameNode));}
   }

   //brandManagerNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* brandManagerNode = xmlNode->FirstChildElement("brandManager");

   if(brandManagerNode){brandManagerIsNull_ = false;}
   else{brandManagerIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- brandManagerNode , address : " << brandManagerNode << std::endl;
   #endif
   if(brandManagerNode)
   {
      if(brandManagerNode->Attribute("href") || brandManagerNode->Attribute("id"))
      {
          if(brandManagerNode->Attribute("id"))
          {
             brandManagerIDRef_ = brandManagerNode->Attribute("id");
             brandManager_ = boost::shared_ptr<CommodityMetalBrandManager>(new CommodityMetalBrandManager(brandManagerNode));
             brandManager_->setID(brandManagerIDRef_);
             IDManager::instance().setID(brandManagerIDRef_,brandManager_);
          }
          else if(brandManagerNode->Attribute("href")) { brandManagerIDRef_ = brandManagerNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { brandManager_ = boost::shared_ptr<CommodityMetalBrandManager>(new CommodityMetalBrandManager(brandManagerNode));}
   }

   //countryNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* countryNode = xmlNode->FirstChildElement("country");

   if(countryNode){countryIsNull_ = false;}
   else{countryIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- countryNode , address : " << countryNode << std::endl;
   #endif
   if(countryNode)
   {
      if(countryNode->Attribute("href") || countryNode->Attribute("id"))
      {
          if(countryNode->Attribute("id"))
          {
             countryIDRef_ = countryNode->Attribute("id");
             country_ = boost::shared_ptr<CountryCode>(new CountryCode(countryNode));
             country_->setID(countryIDRef_);
             IDManager::instance().setID(countryIDRef_,country_);
          }
          else if(countryNode->Attribute("href")) { countryIDRef_ = countryNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { country_ = boost::shared_ptr<CountryCode>(new CountryCode(countryNode));}
   }

   //producerNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* producerNode = xmlNode->FirstChildElement("producer");

   if(producerNode){producerIsNull_ = false;}
   else{producerIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- producerNode , address : " << producerNode << std::endl;
   #endif
   if(producerNode)
   {
      if(producerNode->Attribute("href") || producerNode->Attribute("id"))
      {
          if(producerNode->Attribute("id"))
          {
             producerIDRef_ = producerNode->Attribute("id");
             producer_ = boost::shared_ptr<CommodityMetalProducer>(new CommodityMetalProducer(producerNode));
             producer_->setID(producerIDRef_);
             IDManager::instance().setID(producerIDRef_,producer_);
          }
          else if(producerNode->Attribute("href")) { producerIDRef_ = producerNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { producer_ = boost::shared_ptr<CommodityMetalProducer>(new CommodityMetalProducer(producerNode));}
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<CommodityMetalBrandName> CommodityMetalBrand::getName()
{
   if(!this->nameIsNull_){
        if(nameIDRef_ != ""){
             return boost::shared_static_cast<CommodityMetalBrandName>(IDManager::instance().getID(nameIDRef_));
        }else{
             return this->name_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityMetalBrandName>();
   }
}
boost::shared_ptr<CommodityMetalBrandManager> CommodityMetalBrand::getBrandManager()
{
   if(!this->brandManagerIsNull_){
        if(brandManagerIDRef_ != ""){
             return boost::shared_static_cast<CommodityMetalBrandManager>(IDManager::instance().getID(brandManagerIDRef_));
        }else{
             return this->brandManager_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityMetalBrandManager>();
   }
}
boost::shared_ptr<CountryCode> CommodityMetalBrand::getCountry()
{
   if(!this->countryIsNull_){
        if(countryIDRef_ != ""){
             return boost::shared_static_cast<CountryCode>(IDManager::instance().getID(countryIDRef_));
        }else{
             return this->country_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CountryCode>();
   }
}
boost::shared_ptr<CommodityMetalProducer> CommodityMetalBrand::getProducer()
{
   if(!this->producerIsNull_){
        if(producerIDRef_ != ""){
             return boost::shared_static_cast<CommodityMetalProducer>(IDManager::instance().getID(producerIDRef_));
        }else{
             return this->producer_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<CommodityMetalProducer>();
   }
}
}

File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-com-5-4/CommodityMetalBrand.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class CommodityMetalBrand : ISerialized
    {
        public CommodityMetalBrand(XmlNode xmlNode)
        {
            XmlNode nameNode = xmlNode.SelectSingleNode("name");
            
            if (nameNode != null)
            {
                if (nameNode.Attributes["href"] != null || nameNode.Attributes["id"] != null) 
                {
                    if (nameNode.Attributes["id"] != null) 
                    {
                        nameIDRef_ = nameNode.Attributes["id"].Value;
                        CommodityMetalBrandName ob = new CommodityMetalBrandName(nameNode);
                        IDManager.SetID(nameIDRef_, ob);
                    }
                    else if (nameNode.Attributes["href"] != null)
                    {
                        nameIDRef_ = nameNode.Attributes["href"].Value;
                    }
                    else
                    {
                        name_ = new CommodityMetalBrandName(nameNode);
                    }
                }
                else
                {
                    name_ = new CommodityMetalBrandName(nameNode);
                }
            }
            
        
            XmlNode brandManagerNode = xmlNode.SelectSingleNode("brandManager");
            
            if (brandManagerNode != null)
            {
                if (brandManagerNode.Attributes["href"] != null || brandManagerNode.Attributes["id"] != null) 
                {
                    if (brandManagerNode.Attributes["id"] != null) 
                    {
                        brandManagerIDRef_ = brandManagerNode.Attributes["id"].Value;
                        CommodityMetalBrandManager ob = new CommodityMetalBrandManager(brandManagerNode);
                        IDManager.SetID(brandManagerIDRef_, ob);
                    }
                    else if (brandManagerNode.Attributes["href"] != null)
                    {
                        brandManagerIDRef_ = brandManagerNode.Attributes["href"].Value;
                    }
                    else
                    {
                        brandManager_ = new CommodityMetalBrandManager(brandManagerNode);
                    }
                }
                else
                {
                    brandManager_ = new CommodityMetalBrandManager(brandManagerNode);
                }
            }
            
        
            XmlNode countryNode = xmlNode.SelectSingleNode("country");
            
            if (countryNode != null)
            {
                if (countryNode.Attributes["href"] != null || countryNode.Attributes["id"] != null) 
                {
                    if (countryNode.Attributes["id"] != null) 
                    {
                        countryIDRef_ = countryNode.Attributes["id"].Value;
                        CountryCode ob = new CountryCode(countryNode);
                        IDManager.SetID(countryIDRef_, ob);
                    }
                    else if (countryNode.Attributes["href"] != null)
                    {
                        countryIDRef_ = countryNode.Attributes["href"].Value;
                    }
                    else
                    {
                        country_ = new CountryCode(countryNode);
                    }
                }
                else
                {
                    country_ = new CountryCode(countryNode);
                }
            }
            
        
            XmlNode producerNode = xmlNode.SelectSingleNode("producer");
            
            if (producerNode != null)
            {
                if (producerNode.Attributes["href"] != null || producerNode.Attributes["id"] != null) 
                {
                    if (producerNode.Attributes["id"] != null) 
                    {
                        producerIDRef_ = producerNode.Attributes["id"].Value;
                        CommodityMetalProducer ob = new CommodityMetalProducer(producerNode);
                        IDManager.SetID(producerIDRef_, ob);
                    }
                    else if (producerNode.Attributes["href"] != null)
                    {
                        producerIDRef_ = producerNode.Attributes["href"].Value;
                    }
                    else
                    {
                        producer_ = new CommodityMetalProducer(producerNode);
                    }
                }
                else
                {
                    producer_ = new CommodityMetalProducer(producerNode);
                }
            }
            
        
        }
        
    
        #region Name_
        private CommodityMetalBrandName name_;
        public CommodityMetalBrandName Name_
        {
            get
            {
                if (this.name_ != null)
                {
                    return this.name_; 
                }
                else if (this.nameIDRef_ != null)
                {
                    name_ = IDManager.getID(nameIDRef_) as CommodityMetalBrandName;
                    return this.name_; 
                }
                else
                {
                      return this.name_; 
                }
            }
            set
            {
                if (this.name_ != value)
                {
                    this.name_ = value;
                }
            }
        }
        #endregion
        
        public string nameIDRef_ { get; set; }
        #region BrandManager_
        private CommodityMetalBrandManager brandManager_;
        public CommodityMetalBrandManager BrandManager_
        {
            get
            {
                if (this.brandManager_ != null)
                {
                    return this.brandManager_; 
                }
                else if (this.brandManagerIDRef_ != null)
                {
                    brandManager_ = IDManager.getID(brandManagerIDRef_) as CommodityMetalBrandManager;
                    return this.brandManager_; 
                }
                else
                {
                      return this.brandManager_; 
                }
            }
            set
            {
                if (this.brandManager_ != value)
                {
                    this.brandManager_ = value;
                }
            }
        }
        #endregion
        
        public string brandManagerIDRef_ { get; set; }
        #region Country_
        private CountryCode country_;
        public CountryCode Country_
        {
            get
            {
                if (this.country_ != null)
                {
                    return this.country_; 
                }
                else if (this.countryIDRef_ != null)
                {
                    country_ = IDManager.getID(countryIDRef_) as CountryCode;
                    return this.country_; 
                }
                else
                {
                      return this.country_; 
                }
            }
            set
            {
                if (this.country_ != value)
                {
                    this.country_ = value;
                }
            }
        }
        #endregion
        
        public string countryIDRef_ { get; set; }
        #region Producer_
        private CommodityMetalProducer producer_;
        public CommodityMetalProducer Producer_
        {
            get
            {
                if (this.producer_ != null)
                {
                    return this.producer_; 
                }
                else if (this.producerIDRef_ != null)
                {
                    producer_ = IDManager.getID(producerIDRef_) as CommodityMetalProducer;
                    return this.producer_; 
                }
                else
                {
                      return this.producer_; 
                }
            }
            set
            {
                if (this.producer_ != value)
                {
                    this.producer_ = value;
                }
            }
        }
        #endregion
        
        public string producerIDRef_ { get; set; }
        
    
        
    
    }
    
}

